### DISCUSSION TITLE
Cannot get_ref of std::vector<std::uint8_t>

### DISCUSSION THREAD

**Discussion by AnDevi** [nlohmann/json#3849] [Q&A] [answered]
I want to get reference of my binary date stored in `std::vector<std::uint8_t>`, but code below won't compile with error:

`const auto & data = dictionary.at("data").get_ref<const std::vector<std::uint8_t> &>();`

error: 
`No matching member function for call to 'get_ptr'`

According to documentation binary type should be stored in `std::vector<std::uint8_t>`. I don't know why it can't see `get_ptr` function for this type if it is one of supported types.

Anyone could tell me why it won't compile ?

**Comment by nlohmann** [comment, marked as answer]
Values inside a `json` array are stored as `std::vector<json>`. While you can *convert* them to a `std::vector<std::uint8_t>` (if the values are numbers), you cannot get a reference to that.

**Reply by AnDevi** [reply]
Thank you for your reply. Another solution if I want to get ref to binary values from json, could be usage of `get_binary` instead of `get_ref` ?

**Reply by nlohmann** [reply]
If it is a binary value, then `get_binary` is the way to go (https://json.nlohmann.me/api/basic_json/get_binary/). It also returns a reference.

**Reply by AnDevi** [reply]
Great, thanks for your time !
